code/game/objects/items/weapons/RCD.dm

```diff
@@ -144,239 +218,229 @@ RCD
 		. = 1
 
 	if(href_list["login"])
-		if(istype(usr,/mob/living/silicon))
-			locked = 0
-		else
-			var/obj/item/I = usr.get_active_hand()
-			if(istype(I, /obj/item/pda))
-				var/obj/item/pda/pda = I
-				I = pda.id
-			var/obj/item/card/id/ID = I
-			if(istype(ID) && ID)
-				if(check_access(ID))
-					locked = 0
-				else 
-					to_chat(usr, "<span class='warning'>This ID lacks access.</span>")
-			else
-				to_chat(usr, "<span class='warning'>You can't swipe without your ID in hand.</span>")
+		if(allowed(usr))
+			locked = FALSE
 		. = 1
 
 	if(href_list["logout"])
-		locked = 1
-		. = 1
-
-	if(href_list["toggle_one_access"] && !locked)
-		one_access = !one_access
+		locked = TRUE
 		. = 1
 
-	if(href_list["toggle_access"] && !locked)
-		var/href_access = text2num(href_list["toggle_access"])
-		if(href_access in door_accesses)
-			door_accesses -= href_access
-		else
-			door_accesses += href_access
-		door_accesses_list = list()
-		for(var/access in get_all_accesses())
-			door_accesses_list[++door_accesses_list.len] = list("name" = get_access_desc(access), "id" = access, "enabled" = (access in door_accesses))
-		. = 1
-
-	if(href_list["choice"] && !locked)
-		var/temp_t = sanitize(copytext(input("Enter a custom Airlock Name.","Airlock Name"),1,MAX_MESSAGE_LEN))
-		if(temp_t)
-			door_name = temp_t
-
-/obj/item/rcd/proc/activate()
-	playsound(loc, usesound, 50, 1)
+	if(!locked)
+		if(href_list["toggle_one_access"])
+			one_access = !one_access
+			. = 1
 
+		if(href_list["toggle_access"])
+			var/href_access = text2num(href_list["toggle_access"])
+			if(href_access in door_accesses)
+				door_accesses -= href_access
+			else
+				door_accesses += href_access
+			door_accesses_list = list()
+			for(var/access in get_all_accesses())
+				door_accesses_list[++door_accesses_list.len] = list("name" = get_access_desc(access), "id" = access, "enabled" = (access in door_accesses))
+			. = 1
+
+		if(href_list["choice"])
+			var/temp_t = sanitize(copytext(input("Enter a custom Airlock Name.", "Airlock Name"), 1, MAX_MESSAGE_LEN))
+			if(temp_t)
+				door_name = temp_t
+
+/obj/item/rcd/proc/mode_turf(atom/A, mob/user)
+	if(isspaceturf(A) || istype(A, /obj/structure/lattice))
+		if(useResource(1, user))
+			to_chat(user, "Building Floor...")

```

Comment:
I'm on a naked `to_chat` crusade. Can we keep this & others spanned up relevantly?

Reply:
Our hero! Sparing us from crimes of indecency :rofl: 
Notification'd be the one here, wouldn't it? I'm sure there's dozens by now to choose from

Reply:
I mean, the old code didn't have any spans, but I meant to add them before I got sidetracked with the UI.